crates/factor-outbound-networking/src/lib.rs

```diff
@@ -42,6 +42,12 @@ impl OutboundNetworkingFactor {
     }
 }
 
+pub fn allowed_hosts(component: &AppComponent<'_>) -> anyhow::Result<Vec<String>> {
```

Comment:
Nit: can you move this function down? Having a helper function like this between two `impl` blocks makes the code harder to read IMO. 

Also, can we just add a doc comment for what this function does to make IDE users lives nicer?